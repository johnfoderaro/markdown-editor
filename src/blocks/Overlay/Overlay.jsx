import React from 'react';
import PropTypes from 'prop-types';
import { hot } from 'react-hot-loader';
import styled from 'styled-components';

const Block = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  background-color: rgba(0,0,0,0.75);
  z-index: 10;
  display: flex;
  flex-direction: column;
  justify-content: center;
  text-align: center;
`;

const Overlay = ({ children }) => (
  <Block>
    {children}
  </Block>
);

Overlay.propTypes = {
  children: PropTypes.arrayOf(PropTypes.element).isRequired,
};

export default hot(module)(Overlay);
