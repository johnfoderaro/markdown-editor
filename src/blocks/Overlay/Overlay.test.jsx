import React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';

import Overlay from './Overlay';

let tree;

beforeAll(() => {
  tree = mount(
    <Overlay>
      <h1>one</h1>
      <p>two</p>
    </Overlay>,
  );
});

describe('<Overlay />', () => {
  it('should be defined', () => {
    expect(tree).toBeDefined();
  });
  it('should render correctly', () => {
    expect(tree).toMatchSnapshot();
  });
  it('should render children correctly', () => {
    expect(tree.find('h1').text()).toEqual('one');
    expect(tree.find('p').text()).toEqual('two');
  });
});
