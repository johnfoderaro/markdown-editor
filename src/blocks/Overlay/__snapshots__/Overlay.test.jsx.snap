// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Overlay /> should render correctly 1`] = `
.c0 {
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  background-color: rgba(0,0,0,0.75);
  z-index: 10;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
}

<Overlay>
  <styled.div>
    <StyledComponent
      forwardedClass={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  background-color: rgba(0,0,0,0.75);
  z-index: 10;
  display: flex;
  flex-direction: column;
  justify-content: center;
  text-align: center;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        <h1>
          one
        </h1>
        <p>
          two
        </p>
      </div>
    </StyledComponent>
  </styled.div>
</Overlay>
`;
